### DISCUSSION TITLE
How to set application name from code?

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2254] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/44497354/230781026-072f6517-e1a4-4413-83a7-8e55b23adf31.png)

How to set this fields from code?

**Comment by ganigeorgiev** [comment]
If you need data validations (eg. the value comes from a client request) you can use [`forms.SettingsUpsert`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.14.3/forms#SettingsUpsert):

```go
form := forms.NewSettingsUpsert(app)
form.Meta.AppName = "example"
err := form.Submit()
```

Alternatively, if you don't need field validations (eg. the value is defined in the code and it is trusted) you can call directly `app.Dao().SaveSettings()`:

```go
settings := app.Dao().Settings()
settings.Meta.AppName = "example"
err := app.Dao().SaveSettings(settings)
```

Similar to the above, you can also use the latter as part of a db migration as shown in the example in https://pocketbase.io/docs/migrations/#migrate-create.

btw. you can find all available `Settings` field in https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.14.3/models/settings#Settings

**Reply by Oggy4th** [reply]
@ganigeorgiev 
When I save settings it gives some error.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102e9f9a0]

goroutine 1 [running]:
github.com/pocketbase/pocketbase/daos.(*Dao).ConcurrentDB(...)
        /Users/oggy/go/pkg/mod/github.com/pocketbase/pocketbase@v0.15.3/daos/base.go:66
github.com/pocketbase/pocketbase/daos.(*Dao).DB(...)
        /Users/oggy/go/pkg/mod/github.com/pocketbase/pocketbase@v0.15.3/daos/base.go:58
github.com/pocketbase/pocketbase/daos.(*Dao).ModelQuery(0x0, {0x1040119f0?, 0x140003aa0e0?})
        /Users/oggy/go/pkg/mod/github.com/pocketbase/pocketbase@v0.15.3/daos/base.go:89 +0x40
github.com/pocketbase/pocketbase/daos.(*Dao).ParamQuery(...)
        /Users/oggy/go/pkg/mod/github.com/pocketbase/pocketbase@v0.15.3/daos/param.go:14
github.com/pocketbase/pocketbase/daos.(*Dao).FindParamByKey(0x140007ffe78?, {0x1030ef684, 0x8})
        /Users/oggy/go/pkg/mod/github.com/pocketbase/pocketbase@v0.15.3/daos/param.go:21 +0x58
github.com/pocketbase/pocketbase/daos.(*Dao).SaveParam(0x103ffd740?, {0x1030ef684, 0x8}, {0x103ee7220, 0x14000529200}, {0x1400083ff38, 0x1, 0x140000021a0?})
        /Users/oggy/go/pkg/mod/github.com/pocketbase/pocketbase@v0.15.3/daos/param.go:38 +0x38
github.com/pocketbase/pocketbase/daos.(*Dao).SaveSettings(...)
        /Users/oggy/go/pkg/mod/github.com/pocketbase/pocketbase@v0.15.3/daos/settings.go:62
```

**Reply by ganigeorgiev** [reply]
Are you sure that `SaveSettings` is called after app bootstrap? Could you provide a reproducible code sample?

**Reply by Oggy4th** [reply]
Here is example
```
func Database() error {
	App = pocketbase.New()
	form := forms.NewSettingsUpsert(App)
	form.Meta.AppName = "test"
	form.Meta.AppUrl = "https://test.com"
	form.Meta.HideControls = true
	form.Logs.MaxDays = 14
	err := form.Submit()
	if err != nil {
		return err
	}
	return nil
}
```

**Reply by ganigeorgiev** [reply, marked as answer]
The app needs to be initialized/boostrapped before being able to perform db operations.

I don't have enough context to understand your use case, but you can attach the form submission in a [`app.OnAfterBootstrap()`](https://pocketbase.io/docs/event-hooks/#app-hooks) hook listener OR manually call [`app.Bootstrap()`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.15.3/core#BaseApp.Bootstrap) after `pocketbase.New()`.
